src/SecTester.Scan/Scan.cs

```diff
@@ -96,6 +96,19 @@ public async Task Stop()
     yield return _scanState;
   }
 
+  public async Task Expect(Severity expectation, CancellationToken cancellationToken = default)
+  {
+    using var cancellationTokenSource =
+      CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+
+    cancellationTokenSource.CancelAfter(_options.Timeout);
+
+    var predicate = (Scan _) =>
+      Task.FromResult(_scanState.IssuesBySeverity?.Any(x => x.Number > 0 && x.Type == expectation) ?? false);
```

Comment:
Severity is a threshold describing a range from x to max, the strict equality won't' work as expected. For details see https://github.com/NeuraLegion/sectester-js/blob/7b409959bb74fbdf00a8179517d81c90f5fd3208/packages/scan/src/models/Severity.ts#L7